astropy/io/fits/column.py

```diff
@@ -1423,7 +1423,10 @@ def dtype(self):
                 else:
                     dt = np.dtype((dt.base, dim))
 
-            fields.append((name, dt))
+            if PY2:  # pragma: py2
+                fields.append((name.encode('utf-8'), dt))
```

Comment:
I think my particular case arose when I put `from __future__ import (..., unicode_literals)` incantation in my module. If this fix is rejected, I will still need to do encoding in my module outside of Astropy. But in that case, someone else might have the same problem down the road. So, we have to make a choice.